modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/siege/SiegePreparationTask.java

```diff
@@ -12,11 +12,44 @@ public SiegePreparationTask(Siege siege) {
 
     @Override
     public void run() {
+        /*String playerDisplayName = Bukkit.getPlayer(siege.getPlayerUUID()).getDisplayName();
+        String siegeName = siege.getName();
+        int delayInMinutes = (siege.getDelayBeforeStart() / 60) / 4 * 3;
+        Bukkit.getServer().broadcastMessage(playerDisplayName + " is preparing to siege " + siegeName + "! All players wishing to participate in the defense should head there immediately! Siege will begin in " + delayInMinutes + " minutes");*/
+        int timePassed = (int)(System.currentTimeMillis()/1000) - siege.getStartTime(); //time passed in seconds
+        if ((timePassed == siege.getDelayBeforeStart() / 4 * 3)||
+                (timePassed == siege.getDelayBeforeStart() / 4 * 2)||
+                (timePassed == siege.getDelayBeforeStart() / 4 )||
+                (timePassed == siege.getDelayBeforeStart())){
```

Comment:
The siege task does not occur every second, so the exact value can be skipped. Don't use exact equality.

Reply:
What other alternatives does exist then?